### PR TITLE
add types for 'macaca-circular-json'

### PR DESCRIPTION
**Pull request by zivyangll**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by zivyangll** [comment]
Because the circular-json has stopped maintenance,I forked macaca-circular-json to modify it. And I very much hope to release it as soon as possible. Thank you very much.

File: types/macaca-circular-json/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (DanielRosenwasser) [inline types/macaca-circular-json/tsconfig.json]: Why isn't `strictNullChecks` on?

File: types/macaca-circular-json/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// Type definitions for circular-json 0.4
```
Comment (DanielRosenwasser) [inline types/macaca-circular-json/index.d.ts]: Why is the package `circular-json` but the folder name is `macaca-circular-json`

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]
@sandersn why didn't the bot prevent the inconsistent naming?

File: types/macaca-circular-json/tsconfig.json (hunk 1 above)
Comment (zivyangll) [reply types/macaca-circular-json/tsconfig.json]: It's same with https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/circular-json/tsconfig.json#L9, so I didn't change it.

File: types/macaca-circular-json/index.d.ts (hunk 2 above)
Comment (zivyangll) [reply types/macaca-circular-json/index.d.ts]: > Why is the package `circular-json` but the folder name is `macaca-circular-json`

fixed, thx~

File: types/macaca-circular-json/tsconfig.json (hunk 1 above)
Comment (DanielRosenwasser) [reply types/macaca-circular-json/tsconfig.json]: Everything looks good now except this. If you flip it on I can pull in this change.

**Pull request merged**